### ISSUE TITLE
`ZeroShrinker` is dead code with fragile, untested invariants

### ISSUE THREAD

**Issue by yoshuawuyts** [yoshuawuyts/heckcheck#10]
## Problem

https://github.com/yoshuawuyts/heckcheck/blob/main/src/shrinker.rs#L33-L90

`ZeroShrinker` is private, never constructed anywhere in the crate, not re-exported from `lib.rs`, and has no tests. It should be triggering a `dead_code` warning.

Its invariants are also fragile. `next()` asserts:

```rust
assert_ne!(self.source[self.index], 0);
self.value = self.source[self.index];
self.source[self.index] = self.source[self.index].div(2);
assert_ne!(self.source[self.index], self.value);
```

This relies on `report()` having been called to re-run `find_next_index_to_zeroify()` between every `next()`. Calling `next()` twice in a row after a byte reaches `0` trips the first assertion. Since `Shrink` is a public, user-implementable trait, the expected call protocol between `next` and `report` is not documented anywhere.

Also note the stray `use std::ops::Div;` mid-file (line 33) — `self.source[i] / 2` reads better and needs no import.

## Suggested resolution

Either:

1. Make it `pub`, document it, export it alongside `Shrinker`, and add unit tests covering the halving behaviour and the revert-on-`Pass` path; or
2. Delete it.

Additionally, document the `next`/`report` call protocol on the `Shrink` trait in `src/shrink.rs` so third-party shrinkers can be written correctly.

**Comment by yoshuawuyts** [comment]
Closed by #13
